src/main/java/Duke.java

```diff
@@ -29,42 +28,80 @@ public static void Greet(){
     }
 
     public static void Echo(){
+        boolean continueChat;
         Scanner in = new Scanner(System.in);
-        String text;
-        String order;
-
-        text = in.nextLine();
-        while(!text.equals("bye")){
-            if(text.equals("list"))
-                printList();
-            else if(text.startsWith("done")){
-                order = text.substring(text.indexOf(' ') + 1);
-                doneTask(Integer.parseInt(order) - 1);
-            }else{
-                addTask(text);
+
+        do{
+            // Get query from user
+            String userInput = in.nextLine();
+
+            // Give response
+            continueChat = giveResponse(userInput);
+
+        }while(continueChat);
+
+        in.close();
+    }
+
+    public static boolean giveResponse(String userInput){
+        if(userInput.equals("list")){
+            printList();
+        }else if(userInput.equals("bye")){
+            Exit();
+            return false;
+        }else{
+            int i = userInput.indexOf(" ");
+            String command = userInput.substring(0, i);
+
+            switch (command){
+                case "todo":
+                case "deadline":
+                case "event":
+                    addTask(command, userInput.substring(i + 1));
```

Comment:
I like how all the respective actions based on the different command have their individual function instead of throwing all the code under the case statements.